src/main/java/com/ericsson/commonlibrary/proxy/InterceptableProxyFactory.java

```diff
@@ -122,7 +122,12 @@ void setSuperclass(Class<?> classToIntercept) {
             }
         }
         try {
-            return cc.toClass();
+        	double javaSpecVersion = Double.parseDouble(System.getProperty("java.specification.version"));
+        	if (javaSpecVersion > 1.8) {
```

Comment:
comparisons with a double is not always safe because of the precision of double.
in this case javaSpecVersion  might be 1.799999 or 1.8000001 which would do different things 

it probably safe in this particular instance however :)